asv_bench/benchmarks/io/excel.py

```diff
@@ -35,4 +35,43 @@ def time_write_excel(self, engine):
         writer_write.save()
 
 
+class ReadExcel:
+
+    params = ["xlrd", "openpyxl", "odf"]
+    param_names = ["engine"]
+
+    def _generate_odf(self):
+        doc = OpenDocumentSpreadsheet()
+        table = Table(name="Table1")
+        for row in self.df.values:
+            tr = TableRow()
+            for val in row:
+                tc = TableCell(valuetype='string')
+                tc.addElement(P(text=val))
+                tr.addElement(tc)
+            table.addElement(tr)
+
+        doc.spreadsheet.addElement(table)
+
+        return doc
+
+    def setup(self, engine):
+        self.df = _generate_dataframe()
+
+        self.bio_read = BytesIO()
```

Comment:
To reduce complexity here you can mimic what is done in the Stata benchmark, namely write to a file instead of a BytesIO object